src/backend/joanie/core/api/admin/__init__.py

```diff
@@ -689,6 +691,29 @@ def refund(self, request, pk=None):  # pylint:disable=unused-argument
 
         return Response(status=HTTPStatus.ACCEPTED)
 
+    @extend_schema(
+        request=None,
+        responses={
+            (200, "text/csv"): OpenApiTypes.OBJECT,
+            404: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(methods=["GET"], detail=False)
+    def export(self, request):
+        """
+        Export orders to a CSV file.
+        """
+        queryset = self.filter_queryset(self.get_queryset())
+        serializer = serializers.AdminOrderListExportSerializer(
+            queryset.iterator(), child=self.get_serializer()
+        )
+        now = timezone.now().strftime("%d-%m-%Y_%H-%M-%S")
+        return StreamingHttpResponse(
+            serializer.csv_stream(),
+            content_type="text/csv",
+            headers={"Content-Disposition": f'attachment; filename="orders_{now}.csv"'},
+        )
```

Comment:
Sorry, I should got this one before. IMO for this endpoint, the pagination should be ignored...

Currently, only the result from the current page are exported.